watertap/unit_models/pressure_changer.py

```diff
@@ -11,35 +11,301 @@
+from pyomo.environ import Var, units as pyunits, Expr_if
+
+from enum import Enum, auto
 
 # Import IDAES cores
 from idaes.models.unit_models.pressure_changer import PumpData
 from idaes.core import declare_process_block_class
+from idaes.core.util.constants import Constants
 import idaes.core.util.scaling as iscale
 
 import idaes.logger as idaeslog
 
 _log = idaeslog.getLogger(__name__)
 
 
+class VariableEfficiency(Enum):
+    none = auto()  # default is constant efficiency
+    flow = auto()  # flow-only correlation
+    flow_head = auto()  # flow and head correlation
+
+
 @declare_process_block_class("Pump")
 class PumpIsothermalData(PumpData):
     """
     Standard Isothermal Pump Unit Model Class
     """
 
+    CONFIG = PumpData.CONFIG()
+    CONFIG.declare(
+        "variable_efficiency",
+        ConfigValue(
+            default=VariableEfficiency.none,
+            domain=In(VariableEfficiency),
+            description="Variable pump efficiency flag",
+            doc="""Indicates the relationship used to define pump efficiency
+    **VariableEfficiency.none** - uses default pump efficiency at BEP
+    **VariableEfficiency.flow** - uses an efficiency correlation scaled to the BEP flow rate
+    **VariableEfficiency.flow_head** - uses an efficiency correlation scaled to the BEP flow rate and head
+    """,
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        # ---------------------------------------
+        # Isothermal pump set up
+        # ---------------------------------------
+
+        self.control_volume.del_component(self.control_volume.enthalpy_balances)
+
+        @self.control_volume.Constraint(
+            self.flowsheet().config.time, doc="Isothermal constraint"
+        )
+        def isothermal_balance(b, t):
+            return b.properties_in[t].temperature == b.properties_out[t].temperature
+
+        # ---------------------------------------
+        # Variable efficiency pump set-up
+        # ---------------------------------------
+
+        if self.config.variable_efficiency is not VariableEfficiency.none:
+            # create additional pyomo variables
+            self.bep_flow = Var(
+                initialize=1.0,
+                doc="Best efficiency point flowrate of the centrifugal pump",
+                units=pyunits.m**3 / pyunits.s,
+            )
+
+            self.bep_eta = Var(
+                initialize=self.efficiency_pump[0].value,
+                doc="Best efficiency of the centrifugal pump",
+                units=pyunits.dimensionless,
+            )
+
+            self.flow_ratio = Var(
+                self.flowsheet().time,
+                initialize=1.0,
+                doc="Ratio of pump flowrate to best efficiency point flowrate",
+                units=pyunits.dimensionless,
+            )
+
+            # unfix the default efficiency from the base class
+            self.efficiency_pump.unfix()
```

Comment:
Is it fixed to begin with? I didn't check, but the only way I'd see this happening was if it was a variable that was treated as a parameter.

Reply:
Otherwise, if it isn't fixed within the model already, there would be no need for this line. You could quickly test this by asserting it isn't fixed after uncommenting this line and checking the build.

Reply:
I was using it earlier to initialize `bep_eta` but then I realized all it was doing was passing the value that initialized `efficiency_pump`. I have removed both instances.